packages/grant-explorer/src/store.ts

```diff
@@ -46,19 +46,49 @@ const defaultVotingTokens = Object.fromEntries(
   })
 ) as Record<ChainId, VotingToken>;
 
+function isSameProject(a: CartProject, b: CartProject): boolean {
+  return (
+    a.grantApplicationId === b.grantApplicationId && a.chainId === b.chainId
+  );
+}
+
+function updateOrInsertCartProject(
+  currentProjects: CartProject[],
+  newProject: CartProject
+): CartProject[] {
+  const initialAcc: {
+    projects: CartProject[];
+    hasUpdatedProject: boolean;
+  } = {
+    projects: [],
+    hasUpdatedProject: false,
+  };
+
+  const result = currentProjects.reduce((acc, project) => {
+    if (isSameProject(project, newProject)) {
```

Comment:
it's not clear from the diff but the bug was here:
``` 
if (isSameProject(project, project)) {
```

Reply:
Thanks for pointing out, would have taken me ages to find!